k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/resolver/K8sResourceHandler.kt

```diff
@@ -57,14 +59,26 @@ class K8sResourceHandler (
         resource: Resource<K8sResourceSpec>,
         diff: ResourceDiff<K8sObjectManifest>
     ): List<Task> {
-
         // send a notification on attempt to deploy the image artifact
-        val imageString = find(resource.spec.template.spec as MutableMap<String, Any?>, "image") as String?
-        imageString?.let{
-            log.info("Deploying artifact $it")
-            notifyArtifactDeploying(resource, getTag(it))
+        if (!resource.spec.template.spec.isNullOrEmpty()) {
+            val imageString = find(resource.spec.template.spec as MutableMap<String, Any?>, "image") as String?
+            imageString?.let{
+                log.info("Deploying artifact $it")
+                notifyArtifactDeploying(resource, getTag(it))
+            }
+        }
```

Comment:
same as above for the if block. I know I wrote this bit, and i could have done a better job!
```
val imageString = find(manifest.spec ?: mutableMapOf()) as String?
```

Reply:
Yep will change.